libp2p/Host.cpp

```diff
@@ -170,15 +184,13 @@ void Host::doneWorking()
 	// disconnect pending handshake, before peers, as a handshake may create a peer
 	for (unsigned n = 0;; n = 0)
 	{
-		{
-			Guard l(x_connecting);
+		DEV_GUARDED(x_connecting)
 			for (auto i: m_connecting)
```

Comment:
You should considered iterating over the shared/weak pointers using the reference, not the copy. I.e. `for (auto& i: m_connecting)`. For performance reasons you may not want to temporary bump the shared/weak counter.

There is also "alwyas correct" variant of the for loop: `for (auto&& x: collection)`. That one will be preferred by later C++ standards by adding the syntax sugar in the form of `for(x: collection)`.